### PR TITLE
Replaced Option arguments with Into<Option>

### LINKED ISSUE #83: Make functions that take an optional use Into<Option<T>>
By using a generic impl for `Into<Option<whatever>>` it means we don't need to specify `Some` for a value when passing it in to a function that takes optional values.

This will improve the user's experience a lot.

### PR DESCRIPTION
**Pull request by Akulen**
Fixes #83

### CODE REVIEW

**Comment by Timidger** [comment]
Thanks for the pr!

Looks like the build is failing due to an error in the (incomplete) rootson example. Try to build with `cargo build --all` in the future to ensure that still compiled.

File: src/types/shell/xdg_shell_v6.rs [hunk 1]
```diff
@@ -84,7 +85,7 @@ impl XdgV6ShellSurface {
                                               } });
         (*shell_surface).data = Box::into_raw(shell_state) as *mut _;
         XdgV6ShellSurface { liveliness: Some(liveliness),
-                            state,
+                            state: state,
```
Comment (Timidger) [inline src/types/shell/xdg_shell_v6.rs]: No need to write out `state: state` explicitly, use the feature I use everywhere else where you can just use the variable name.

Comment (Akulen) [reply src/types/shell/xdg_shell_v6.rs]: Yeah, I had to use state.into(), but as state is cloned before that, i opted to replace state with state.into() at the start of the function

**Comment by Timidger** [comment]
Thanks! I'm going to merge this as is, but you should consider signing your commits with a GPG key in the future.

If you are looking for meatier work, ping me on gitter or freenode some time and I can walk you through working on either Way Cooler or the Rust version of Rootston. That's really fun because then that's actually working on a compositor rather than a bindings library :)

**Pull request merged**